File: challenging/3_channels/channels.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func CalculateValue(values chan int) {
	value := rand.Intn(10)
	fmt.Println("Calculated Random Value: {}", value)
	values <- value
}

func main() {
	rand.Seed(time.Now().UnixNano())
	fmt.Println("Go Channel Tutorial")

	values := make(chan int)
	defer close(values)

	go CalculateValue(values)

	value := <-values
	fmt.Println(value)
}

File: challenging/3_channels/channels.js
function CalculateValue(values) {
	value = Math.random() * 10;
	console.log("Calculated Random Value: ", value);
	values.push(value);
}

function main() {
	console.log("Go Channel Tutorial");

	values = [];

	CalculateValue(values);

	value = values.pop();
	console.log(value);
}

main();